File: laptop-store-backend/model/src/main/java/org/example/model/entity/user/UserLink.java
package org.example.model.entity.user;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "user_link")
@Data
public class UserLink implements Serializable {
    @Id
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "facebook_id", length = 30)
    private String facebookId;

    @Column(name = "google_id", length = 30)
    private String googleId;
}

File: laptop-store-backend/model/src/main/kotlin/org/example/model/entity/user/UserLink.kt
package org.example.model.entity.user

import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "user_link")
class UserLink : Serializable {
    companion object {
        private const val serialVersionUID: Long = 1L
    }

    @Id
    @OneToOne
    @JoinColumn(name = "user_id")
    val user: User? = null

    @Column(name = "facebook_id", length = 30)
    val facebookId: String? = null

    @Column(name = "google_id", length = 30)
    val googleId: String? = null
}